### DISCUSSION TITLE
Separate configs for local devcontainer and codespaces

### DISCUSSION THREAD

**Discussion by 0x77dev** [community/community#5580] [Codespaces]
Would be nice to have the possibility to create a `.devcotnainer` folder separate for local development and codespaces
for example use `.codespace` and fallback to `.devcontainer` in GitHub Codespaces
or specify the folder in some kind of config file

Codespaces does not support mounts in docker-compose and it crashes, but it necessary for local development

**Comment by GLStephen** [comment]
Might be nice to split .devcontainer into:

```
local: { config: "for local" },
codespace: { config: "for codespace" }
```

**Comment by mestir** [comment]
Le dim. 12 sept. 2021 ? 02:52, Stephen Johnston ***@***.***> a ?crit :
Might be nice to split .devcontainer into:
local: { config: "for local" },
codespace: { config: "for codespace" }


?You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by 2percentsilk** [comment]
Hi @0x77dev! Can you elaborate a little more on why you'd need separate configs for local versus codespaces? Is it primarily the mount issue with docker-compose?

**Comment by GLStephen** [comment]
@2percentsilk there may be local resources not available from the codespace, and we could have technical and philosophical debates about that, but it happens. Particularly in terms of deploying locally to access a private or secure resource. Some amount of "configure local this way" and "configure codespaces this way" seems sensible. Even if it were a way to detect the environment and do it via app config versus actual bifurcated configs.
